### PR TITLE
Unification of naming

### LINKED ISSUE #5285: Unification of naming, number of arguments and returning properties in core interfaces
## What is the motivation for adding / enhancing this feature?
Sometimes we have discrepancies between naming convention or number of arguments given to the very similar functions

## What are the acceptance criteria 
- [x] each composable, should return the following fields:  
- the one with the response such as cart, products, categories, etc.
- error, loading - for the current state
- all of the required functions
- if the response contains a collection of the data, please introduce getter `getItems` (example: cart, billing, product, wishlist, and more)

any already existing field should be removed and reading of the corresponding field should be moved to the getter

- [x] each factory param function should have always 2 arguments:
- first one is the context
- second one is the real params of the function
- customQuery should be a part of the params

- [x] rename each `loadXXX` function to just `load`
- [x] rename `setup` to `provide` in factoryParams

## Version of Vue Storefront

- [ ] Vue Storefront
- [x] Vue Storefront Next

## Can you complete this feature request by yourself?

- [x] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by Fifciu**
### Related Issues

closes (partially) #5285

### Additional info
Probably, we should also check Wishlist in enterprise.

### Which Environment This Relates To
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [x] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog
- [ ] No upgrade steps required (100% backward compatibility and no breaking changes)
- [x] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing Vue Storefront sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: packages/core/nuxt-theme-module/theme/components/CartSidebar.vue [hunk 1]
```diff
@@ -136,14 +136,14 @@ export default {
   },
   setup() {
     const { isCartSidebarOpen, toggleCartSidebar } = useUiState();
-    const { cart, removeFromCart, updateQuantity, loadCart } = useCart();
+    const { cart, removeFromCart, updateQuantity, load } = useCart();
```
Comment (andrzejewsky) [inline packages/core/nuxt-theme-module/theme/components/CartSidebar.vue:139]: please do a renaming everywhere: `load: loadCart` instead - by doing this we'll see what exactly we are loading here, that's why we wanted to rename it to just load everywhere

**Review by andrzejewsky — CHANGES_REQUESTED** [review]

**Review by filrak — APPROVED** [review]

**Review by filrak — CHANGES_REQUESTED** [review]
Is this change needed in factory params? iI think it's unnecessary there to change the naming and we can avoid this additional BC for integrators and focus on public API - WDYT ?

File: packages/boilerplate/theme/pages/Checkout/Payment.vue [hunk 2]
```diff
@@ -246,7 +246,7 @@ export default {
 
     onMounted(async () => {
       if (isAuthenticated.value) {
-        await loadBilling();
+        await loadUserBilling();
```
Comment (filrak) [inline packages/boilerplate/theme/pages/Checkout/Payment.vue:249]: shouldn't that be `load` ?

**Review by andrzejewsky — APPROVED** [review]

**Review by filrak — CHANGES_REQUESTED** [review]
you havn't mentioned the custom query change. it's also worth supporting the additional argument for some time and mark it as deprecated as well

File: packages/core/core/src/factories/useCartFactory.ts [hunk 3]
```diff
@@ -113,12 +119,16 @@ export const useCartFactory = <CART, CART_ITEM, PRODUCT, COUPON>(
         return;
       }
       loading.value = true;
-      cart.value = await factoryParams.loadCart(context, customQuery);
+      cart.value = await markMethodDeprecated(
+        '\'loadCart\' is deprecated, use \'load\' in your integration instead',
```
Comment (filrak) [inline packages/core/core/src/factories/useCartFactory.ts:123]:
```suggestion
        '\'loadCart\' is deprecated and will be removed in version 2.1.0, use \'load\' in your integration instead. Read more: https://github.com/DivanteLtd/vue-storefront/pull/5299/',
```

Comment (filrak) [reply packages/core/core/src/factories/useCartFactory.ts:123]: please correct all depreciation messages accordingly so the users see
- how long they can use the old api
- some reference to the context of change

File: packages/core/docs/composables/use-wishlist.md [hunk 4]
```diff
@@ -18,11 +18,11 @@ export default {
     const { 
       wishlist,
       addToWishlist,
-      loadWishlist,
+      load,
```
Comment (filrak) [inline packages/core/docs/composables/use-wishlist.md:21]: what does this `load` do?

File: packages/core/nuxt-theme-module/theme/components/AppHeader.vue [hunk 5]
```diff
@@ -46,9 +46,9 @@ export default {
   setup(props, { root }) {
     const { toggleCartSidebar, toggleWishlistSidebar, toggleLoginModal } = useUiState();
     const { changeSearchTerm, getFacetsFromURL } = useUiHelpers();
-    const { isAuthenticated, load } = useUser();
-    const { cart, loadCart } = useCart();
-    const { loadWishlist } = useWishlist();
+    const { isAuthenticated, load: loadUser } = useUser();
```
Comment (filrak) [inline packages/core/nuxt-theme-module/theme/components/AppHeader.vue:49]: how about something like
```js
const $user = useUser()
const $cart = useCart()
// later
$user.load()
$cart.load()
```

Subjectively it's cleaner and can be a convention but I'm curious about your opinion

**Comment by Fifciu** [comment]
| Composable | Method | Old signature | New signature |
|------------|--------|---------------|---------------|
|      useCart      |   addToCart     |       (product: PRODUCT, quantity: number, customQuery?: CustomQuery)        |       ({ product: PRODUCT, quantity: number, customQuery?: CustomQuery })        |
|      useCart      |   removeFromCart    |       (product: CART_ITEM, customQuery?: CustomQuery)        |       ({ product: CART_ITEM, customQuery?: CustomQuery })        |
|      useCart      |   updateQuantity     |       (product: CART_ITEM, quantity?: number, customQuery?: CustomQuery)        |    ({ product: CART_ITEM, quantity?: number, customQuery?: CustomQuery })           |
|      useCart      |   load     |      (customQuery?: CustomQuery)         |        ({ customQuery?: CustomQuery } = {})       |
|      useCart      |   clearCart    |      ()         |       ()        |
|      useCart      |   applyCoupon     |      (couponCode: string, customQuery?: CustomQuery)         |        ({ couponCode: string, customQuery?: CustomQuery })       |
|      useCart      |   isOnCart    |      (product: PRODUCT)         |       ({ product?: PRODUCT })        |
|      useCart      |   removeCoupon     |       (coupon: COUPON, customQuery?: CustomQuery)        |       ({ coupon: COUPON, customQuery?: CustomQuery })        |
|      useCart      |   setCart     |       (newCart: CART)        |       ({ newCart?: CART })        |

File: packages/core/nuxt-theme-module/theme/components/AppHeader.vue (hunk 5 above)
Comment (Fifciu) [reply packages/core/nuxt-theme-module/theme/components/AppHeader.vue:49]: I like this part. What about you @andrzejewsky ?

**Comment by Fifciu** [comment]
| Factory | Old method | New method | Old signature | New signature |
|------------|--------|---------------|---------------|---|
|      useCartFactory      |   addToCart     |  addItem |     context, { currentCart: cart.value, product, quantity }, customQuery      |       context, { currentCart: cart.value, product, quantity, customQuery }      |

**Review by andrzejewsky — APPROVED** [review]

**Pull request merged**